### PR TITLE
Check the sensibility of the results

### PR DESCRIPTION
**Pull request by Eleonore9**
Imitate the soak test in witan.r.models to be able to run the model for all English local authorities and check the values are sensible.

* Look for negative households values
* Check households, total households and dwellings are always positive values
*Note: households can be equal to zero*
* Run the test once and comment out as it takes almost 20 minutes to run
* Debug issues in the test and fix the code to cut off data before ONS input start date

### CODE REVIEW

File: test/witan/models/soak_test.clj [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+             [witan.workspace-api.protocols :as p]
+             [witan.workspace-executor.core :as wex]
+             [clojure.core.matrix.dataset :as ds]
+             [clojure.data.csv :as data-csv]
+             [schema.coerce :as coerce]
+             [witan.datasets :as wds]
+             [clojure.java.io :as io]
+             [clojure.edn :as edn]))
+
+(defn- load-csv
```
Comment (acron0) [inline test/witan/models/soak_test.clj]: This fn appears in more than one place - is it able to be moved to another location?

EDIT: Looks this this is true of a few  of the other fns in this file

File: test/witan/models/soak_test.clj [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+(defn spot-negative [[gss-code result]]
+  (when (some neg? (-> result
+                       :households
+                       (wds/subset-ds :cols :households)))
+    (println (name gss-code) (get english-local-authorities gss-code))))
+
+;; The following test runs on all English local authorities in over 18 minutes
+;; You must have run `lein split-data` on the command-line before
+
+#_(deftest all-local-authorities-test
```
Comment (acron0) [inline test/witan/models/soak_test.clj]: It'd be good if you could toggle this test on with the presence of an environmental var, then we can have it run on Circle CI - but I understand not wanting to run it locally every time, if it takes 18 mins

**Review by acron0 — CHANGES_REQUESTED** [review]

File: test/witan/models/soak_test.clj (hunk 1 above)
Comment (Eleonore9) [reply test/witan/models/soak_test.clj]: @acron0: You tend not to recommend defining a set of helper fns in a `utils.clj` file.
That'd be the only alternative I'd think of here!

File: test/witan/models/soak_test.clj [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+;; Test the model can run in the workspace for all English local authorities
+
+(defn with-gss
+  [id gss-code]
+  (str id "_" gss-code ".csv"))
+
+(defn local-inputs
+  [gss-code]
+  { :population [(with-gss "./data/default_datasets/population/ons_2014_based_snpp" gss-code)
```
Comment (sunnytownsend) [inline test/witan/models/soak_test.clj]: white space

Comment (sunnytownsend) [inline test/witan/models/soak_test.clj]: Have you run eastwood etc?

**Comment by sunnytownsend** [comment]
have you run eastwood etc?

**Comment by sunnytownsend** [comment]
great job Eleonore here is your model review :shipit: (pending tiny comments)

**Comment by Eleonore9** [comment]
Ran `lein eastwood` and `lein kibit` -> no changes
Ran `lein cljfmt check/fix` -> formatted several namespaces

File: project.clj [hunk 4]
```diff
@@ -7,7 +7,8 @@
                  [net.mikera/core.matrix "0.55.0"]
                  [witan.workspace-api "0.1.20"]
                  [prismatic/schema "1.1.3"]
-                 [org.clojure/data.csv "0.1.3"]]
+                 [org.clojure/data.csv "0.1.3"]
+                 [environ "1.0.1"]]
```
Comment (acron0) [inline project.clj]: Could you move this into the dev dependencies?

**Review by acron0 — DISMISSED** [review]

**Pull request merged**